easybuild/easyblocks/m/molpro.py

```diff
@@ -0,0 +1,196 @@
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Molpro, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import fileinput
+import os
+import re
+import sys
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import mkdir, read_file
+from easybuild.tools.run import run_cmd
+
+
+class EB_Molpro(ConfigureMake):
+    """Support for building and installing Molpro."""
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, initialize class variables specific to Molpro and check on license token."""
+        super(EB_Molpro, self).__init__(*args, **kwargs)
+
+        self.full_prefix = None
+        self.orig_launcher = None
+
+        self.cleanup_token_symlink = False
+        self.license_token = os.path.join(os.path.expanduser('~'), '.molpro', 'token')
+
+        if not os.path.isfile(self.license_token):
+            if self.cfg['license_file'] is not None and os.path.isfile(self.cfg['license_file']):
+                # put symlink in place to specified license file in $HOME/.molpro/token
+                # other approaches (like defining $MOLPRO_KEY) don't seem to work
+                self.cleanup_token_symlink = True
+                mkdir(os.path.dirname(self.license_token))
+                try:
+                    os.symlink(self.cfg['license_file'], self.license_token)
+                    self.log.debug("Symlinked %s to %s", self.cfg['license_file'], self.license_token)
+                except OSError, err:
+                    raise EasyBuildError("Failed to create symlink for license token at %s", self.license_token)
+
+            else:
+                # no license token available, so fail early
+                raise EasyBuildError("No license token found at either %s or via 'license_file'", self.license_token)
+
+    def configure_step(self):
+        """Custom configuration procedure for Molpro: use 'configure -batch'."""
+
+        if not self.cfg['license_file'] or not os.path.exists(self.cfg['license_file']):
```

Comment:
Yeah, also, I had to move this to the `configure_step`, for compatibility with `--module-only`.